.github/workflows/dependency-check.yml

```diff
@@ -0,0 +1,101 @@
+name: Dependency Check
+
+on:
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'go.sum'
+      - '.github/workflows/dependency-check.yml'
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-latest
+    name: Dependency Check
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - id: dependencies
+        env:
+          ALLOWED: |
+            [
+              {
+                "Path": "github.com/filecoin-project/go-data-transfer/v2",
+                "Version": "v2.0.0-rc7",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/filecoin-project/go-state-types",
+                "Version": "v0.16.0-dev",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/syndtr/goleveldb",
+                "Version": "v1.0.1-0.20210819022825-2ae1ddf74ef7",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/xorcare/golden",
+                "Version": "v0.6.1-0.20191112154924-b87f686d7542",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/xordataexchange/crypt",
+                "Version": "v0.0.3-0.20170626215501-b2862e3d0a77",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/yugabyte/pgx/v5",
+                "Version": "v5.5.3-yb-2",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "go.dedis.ch/kyber/v4",
+                "Version": "v4.0.0-pre2.0.20240924132404-4de33740016e",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "gopkg.in/check.v1",
+                "Version": "v1.0.0-20201130134442-10cb98267c6c",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "gopkg.in/tomb.v1",
+                "Version": "v1.0.0-20141024135613-dd632973f1e7",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "honnef.co/go/tools",
+                "Version": "v0.0.1-2020.1.4",
+                "Reason": "unknown"
+              },
+              {
+                "Path": "github.com/quic-go/webtransport-go",
+                "Version": "v0.8.1-0.20241018022711-4ac2c9250e66",
+                "Reason": "unknown"
+              }
+            ]
+        run: |
+          echo "unreleased<<EOF" >> $GITHUB_OUTPUT
+          go list -m -json all |
+            jq -s '
+              (
+                map({Path: .Path, Version: .Version}) |
+                map(select(.Version | test("^v\\d+\\.\\d+\\.\\d+-")?)) |
+                map(select(.Version | test("^v0\\.0\\.0-")? | not))
```

Comment:
Go newbie question: is it possible to have a `v0.0.0` dependency if the the dependended-module actually has done a release?

Basically, per [here](https://github.com/filecoin-project/lotus/issues/7131#issuecomment-2359417825) we're excluding 100+ dependencies because they are v0.0.0 in our `go.mod`, but is it possible that we should actually be depending on non-v0.0.0 versions for some of them now?  My worry is that v0.0.0 dependencies get to fly under the radar and there is no forcing function to get them to "fly above the radar".

Reply:
@rvagg : can you speak to this comment?  (I don't know the Go world to know if this is a problem.)

Reply:
I'm pretty sure it's possible although I can't point to an instance. It's one of the reasons I was suggesting we be more specific in the new form @ https://github.com/filecoin-project/lotus/pull/12740#discussion_r1877013484 - regex the `v0.0.0-date-commit` form and avoid any other variations.

Reply:
This is resolved by https://github.com/filecoin-project/lotus/pull/12774 .  That PR checks that any v0.0.0 dependency doesn't have a tagged version and if it doesn, it must have the `dependency-check-ignore`.